### PR TITLE
Implement Links and Refs

### PR DESCRIPTION
**Pull request by brendanzab**
This is important for parsing formats like OpenType that store references to other parts of the binary stream in their structure.

### CODE REVIEW

**Comment by brendanzab** [comment]
Making progress! Just need to add some tests.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        rust: ["1.42.0", "stable", "beta", "nightly"]
+        rust: ["stable"]
```
Comment (toothbrush) [inline .github/workflows/ci.yml:67]: This seems like a reasonable efficiency gain :ok_hand:

Comment (brendanzab) [reply .github/workflows/ci.yml:67]: thankfully github's parallism seems pretty great, but yeah, also means we are less stuck if rustfmt changes! 😆

**Comment by brendanzab** [comment]
Any qualms about me merging this?

**Comment by mikeday** [comment]
Looks good to me! :nerd_face:

**Comment by toothbrush** [comment]
> Any qualms about me merging this?

Sounds exciting!

**Pull request merged**